spec/mail/messagelist.mdown

```diff
@@ -89,6 +89,8 @@ A **FilterCondition** object has the following properties:
   Looks for the text in the *subject* property of the message.
 - **body**: `String|null`
   Looks for the text in the *textBody* or *htmlBody* property of the message.
+- **attachments**: `String|null`
+  Looks for the text in the attachments of the message. Server MAY handle text extraction when possible for the different kinds of media.
```

Comment:
This should probably be searched automatically if you search with the "text" property too (or maybe we redefine that to simply be more generic, so we don't have to explicitly list all the others it matches).

Reply:
Actually I believe having the semantic available to allow explicitly looking up in attachments is a bonus. It allows to add such features like:

<<Oh, I need to retrieve the email with the odt document about last strategy meeting>>

Is clearly more specific than:

<<Give me all the mail about strategy meeting>>

The idea you gave (weakening text search) actually can be emulated with a AND between text and attachment search.

More generally, I would value having searches close to the message object.

Reply:
Is clearly more specific than:

Search all the emails that contains 'strategy meeting'

Sorry for the display issue.

Reply:
No, you misunderstood my comment. I'm not suggesting removing the *attachments* property for explicitly searching within attachments.

The *text* property is a shorthand for searching all the text-based properties (to, body etc.), and so I'm saying as *attachments* is a text-based search, this should also be referenced in the list of properties that *text* searches, or probably better update the definition of *text* so it automatically includes all text-based properties added in the future.

Reply:
Q for Neil: do you want 'text' search to cover all future non-mutable text-based searches, or all future text based searches. This detail matters when annotation support (RFC 5257) is added.

Reply:
Ok, yes, sure. I will update the proposal to match this.

Reply:
cjnewman: Good question. I think annotations should probably always be explicit, as whether they make sense to include depends on the semantics the annotation is being used for.